### PR TITLE
Added Flask to opencv install script and created basic app

### PR DESCRIPTION
**Pull request by MicLi01**
Added Flask to Opencv install script.
Created basic opencv app that can display a video.

### CODE REVIEW

File: opencv_flask/app.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from flask import Flask, request
+import cv2
+import numpy as np
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    return 'Hello World!'
+
+@app.route('/display')
```
Comment (chc5) [inline opencv_flask/app.py]: Comment what which method does.

File: opencv_flask/app.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+from flask import Flask, request
+import cv2
+import numpy as np
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
```
Comment (chc5) [inline opencv_flask/app.py]: Comment please

File: opencv_flask/app.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    return 'Hello World!'
+
+@app.route('/display')
+def display_vid():
+    file_name = request.args.get('file_name', default = '', type = str)
+    cap = cv2.VideoCapture(file_name)
+    if (cap.isOpened()== False): 
+        print("Error opening video stream or file")
```
Comment (chc5) [inline opencv_flask/app.py]: Give some kind of information on where this error is coming from. Judging by this if statement, it seems like it's coming from cv2

File: opencv_flask/app.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        # Break the loop
+        else: 
+            break
+ 
+    # When everything done, release the video capture object
+    cap.release()
+ 
+    # Closes all the frames
+    cv2.destroyAllWindows()
+
```
Comment (chc5) [inline opencv_flask/app.py]: Print out to the console that it stopped recording.

File: opencv_flask/app.py (hunk 1 above)
Comment (MicLi01) [reply opencv_flask/app.py]: done

File: opencv_flask/app.py (hunk 4 above)
Comment (MicLi01) [reply opencv_flask/app.py]: Done

File: opencv_flask/app.py (hunk 1 above)
Comment (MicLi01) [reply opencv_flask/app.py]: done

File: opencv_flask/app.py (hunk 2 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed this

File: opencv_flask/app.py (hunk 3 above)
Comment (MicLi01) [reply opencv_flask/app.py]: added. cv2 is opencv.

File: opencv_flask/app.py (hunk 4 above)
Comment (MicLi01) [reply opencv_flask/app.py]: added

File: scripts/lib/install_opencv.sh [hunk 5]
```diff
@@ -1,13 +1,34 @@
...
+  sudo apt install python3-venv
+  
+  # Change To Opencv server directory
+  cd ../opencv_flask
+
+  # Create new virtual environment
+  python3 -m venv venv
+  
+  # Running the activate script
+  source venv/bin/activate
```
Comment (chc5) [inline scripts/lib/install_opencv.sh]: I recommend doing this approach from now on when installing pip packages and not using shell script to install them individually: https://pip.pypa.io/en/stable/reference/pip_freeze/

File: opencv_flask/app.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
+from flask import Flask
+import cv2
+import numpy as np
+app = Flask(__name__)
+
+def display_vid(file_name):
```
Comment (chc5) [inline opencv_flask/app.py]: There's no need to comment inside the body of the method unless it's unclear by just reading it. Comment just on top of the method that explicitly tells us what this method do in a brief and concise manner.

File: opencv_flask/app.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+        ret, frame = cap.read()
+        if ret == True:
+            # Display the resulting frame
+            cv2.imshow('Frame',frame)
+
+            # Press Q on keyboard to exit
+            if cv2.waitKey(25) & 0xFF == ord('q'):
+                break
+
+        # Break the loop
```
Comment (chc5) [inline opencv_flask/app.py]: Unnecessary comment

File: scripts/lib/install_opencv.sh [hunk 8]
```diff
@@ -1,13 +1,34 @@
 #
-# Install opencv and its dependencies
+# Install opencv and related dependencies
 
 install_opencv() {
```
Comment (chc5) [inline scripts/lib/install_opencv.sh]: We dont need comments for each command unless you want to group commands under one comment

File: opencv_flask/app.py [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        else: 
+            break
+ 
+    # When everything done, release the video capture object
```
Comment (chc5) [inline opencv_flask/app.py]: Unnecessary comment

File: opencv_flask/app.py [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    cap.release()
+ 
+    # Closes all the frames
```
Comment (chc5) [inline opencv_flask/app.py]: Unnecessary comment

File: opencv_flask/app.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+    if (cap.isOpened()== False): 
+        print("OpenCV failed to open video stream or file")
+    while(cap.isOpened()):
+        # Capture frame-by-frame
+        ret, frame = cap.read()
+        if ret == True:
+            # Display the resulting frame
+            cv2.imshow('Frame',frame)
+
+            # Press Q on keyboard to exit
```
Comment (chc5) [inline opencv_flask/app.py]: Necessary comment

File: opencv_flask/app.py [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+def display_vid(file_name):
+    # Displays a video from the file_name query parameter
+    cap = cv2.VideoCapture(file_name)
+    if (cap.isOpened()== False): 
+        print("OpenCV failed to open video stream or file")
+    while(cap.isOpened()):
+        # Capture frame-by-frame
+        ret, frame = cap.read()
+        if ret == True:
+            # Display the resulting frame
```
Comment (chc5) [inline opencv_flask/app.py]: Unnecessary comment

File: opencv_flask/app.py [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+import numpy as np
+app = Flask(__name__)
+
+def display_vid(file_name):
+    # Displays a video from the file_name query parameter
+    cap = cv2.VideoCapture(file_name)
+    if (cap.isOpened()== False): 
+        print("OpenCV failed to open video stream or file")
+    while(cap.isOpened()):
+        # Capture frame-by-frame
```
Comment (chc5) [inline opencv_flask/app.py]: Unnecessary comment

File: scripts/lib/install_opencv.sh (hunk 5 above)
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: Created a new opencv_requirements file

File: opencv_flask/app.py (hunk 6 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: opencv_flask/app.py (hunk 7 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: scripts/lib/install_opencv.sh (hunk 8 above)
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: removed comments

File: opencv_flask/app.py (hunk 10 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: opencv_flask/app.py (hunk 9 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: opencv_flask/app.py (hunk 11 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: opencv_flask/app.py (hunk 12 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: opencv_flask/app.py (hunk 13 above)
Comment (MicLi01) [reply opencv_flask/app.py]: removed comment

File: scripts/lib/install_opencv.sh [hunk 14]
```diff
@@ -1,13 +1,23 @@
...
+  
+  cd ../src/cv/
+
+  # Setting up new virtual environment
+  python3 -m venv venv
+  
+  source venv/bin/activate
+
+  # Install opencv requirements
+  pip install -r opencv_requirements.txt
```
Comment (chc5) [inline scripts/lib/install_opencv.sh]: Change this filename to requirements.txt

File: src/cv/test_app.py [hunk 15]
```diff
@@ -0,0 +1,14 @@
+import mock
```
Comment (chc5) [inline src/cv/test_app.py]: Rename this to tests.py

File: scripts/lib/install_opencv.sh (hunk 14 above)
Comment (MicLi01) [reply scripts/lib/install_opencv.sh]: changed

File: src/cv/test_app.py (hunk 15 above)
Comment (MicLi01) [reply src/cv/test_app.py]: changed

**Comment by chc5** [comment]
lgtm

**Comment by melvinheks** [comment]
Nice

**Pull request merged**